iphone/Maps/Classes/CarPlay/CarPlayService.swift

```diff
@@ -60,6 +60,33 @@ final class CarPlayService: NSObject {
     FrameworkHelper.updatePositionArrowOffset(false, offset: 5)
   }
 
+  private var savedInterfaceController: CPInterfaceController?
+  @objc func showOnPhone() {
+    savedInterfaceController = interfaceController
+    destroy()
+    showPhoneModeAlert()
+  }
+
+  @objc func showOnCarplay() {
+    if let window, let savedInterfaceController {
+      setup(window: window, interfaceController: savedInterfaceController)
+    }
+  }
+
+  private func showPhoneModeAlert() {
+    let alert = CPAlertTemplate(
+      titleVariants: [L("carplay_used_on_the_phone_screen")],
+      actions: [.init(title: L("carplay_continue_in_the_car"),
```

Comment:
Please avoid using the `.init(...` because it hides the object type on creation and makes it hard to find such objects.